powersimdata/input/grid.py

```diff
@@ -27,9 +31,12 @@ def __init__(self, interconnect, source="usa_tamu", engine="REISE"):
         """Constructor."""
         if not isinstance(source, str):
             raise TypeError("source must be a str")
-        supported_engines = {"REISE", "REISE.jl"}
-        if engine not in supported_engines:
-            raise ValueError(f"Engine must be one of {','.join(supported_engines)}")
+        if source not in self.SUPPORTED_MODELS:
```

Comment:
I believe `source` can also be a path to a `.mat` file which contains a modified grid. See L#47 where this is handled.

Reply:
@jon-hagg updated the input validation to account for this. I think long term though the `source` parameter should be refactored to be an object that's passed into this class. In other words, this logic:
```
        if cached is not None:
            data = cached
        elif source == "usa_tamu":
            data = TAMU(interconnect)
        elif os.path.splitext(source)[1] == ".mat":
            if engine == "REISE":
                data = FromREISE(source)
            elif engine == "REISE.jl":
                data = FromREISEjl(source)
```
should exist outside of the Grid class. This makes the code more maintainable by using [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). As the code base grows, adding more source/engine options and tests to this class is going to become more of a PITA. Thoughts on adding a github issue/jira ticket/asana task (not sure how action items are tracked 😅 ) for refactoring this Grid class?

Reply:
A github issue is probably best. This idea doesn't quite fit into our definition of a 'feature', but I think the feature request issue template is probably the best to use for a refactor like this.

Reply:
We also need to factor out a lot of code from the `TAMU` class since other grid models will have the same input files (CSV) with the same columns and header.

Reply:
Part of that work was done in my branch `daniel/read_hifld_grid`, which is behind `develop` by a bit, but most of the relevant commits should be cherry-pickable without any merge conflicts.